sys/dev/drm/panfrost/panfrost_gem.c

```diff
@@ -689,10 +695,13 @@ panfrost_gem_get_pages(struct panfrost_gem_object *bo)
 	bo->pages = m0;
 	bo->npages = npages;
 
-	if (1 == 0)
+#ifdef IOMMU
+	if (1 == 1)
 		error = panfrost_alloc_pages_iommu(bo);
 	else
+#else
 		error = panfrost_alloc_pages_contig(bo);
+#endif
 
 	if (error)
```

Comment:
This is broken, with IOMMU enabled it parses as:
```
if (1 == 1)
  error = panfrost_alloc_pages_iommu(bo);
else if (error)
  return (error);
```
which is nonsense. I don't understand why the `if (1 == 1)` is there rather than entirely `#ifdef`.